build-tools/xaprepare/xaprepare/Application/Program.ArchLinux.cs

```diff
@@ -1,29 +1,139 @@
 using System;
+using System.Collections.Generic;
+using System.IO;
 using System.Threading.Tasks;
 
 namespace Xamarin.Android.Prepare
 {
-	class ArchLinuxProgram : LinuxProgram
-	{
-		public ArchLinuxProgram (string packageName, string executableName = null)
-			: base (packageName, executableName)
-		{}
+    class ArchLinuxProgram : LinuxProgram
+    {
+        private static string aurHelper = null;
+		// TODO: check if binary name is right.
+        private static string[] aurHelpersToCheck = new string[] { "yay","aurman","pacaur","pakku","aura","pikaur","trizen","pbget","auracle","repofish","pb", "bb", "rua" };
```

Comment:
Can you explain, in a comment here, what are these programs for?

Reply:
Arch Linux has two sources for installing packages, first is through "pacman" and repositories containing prebuilt packages, the other is "Arch User Repository" (AUR) which contains PKGBUILD files for building packages from scratch and then installing them. To install Oracle's JDK, you should use its AUR package, as it is not in the official repositories. To ease usage of AUR, many "AUR Helper"s have been developed and their CLI arguments are merely the same as pacman's. I decided to use them instead of pacman when available for package installation. 6 hours ago I thought it would be a great thing, but now I think I should remove it and leave it to user to install those custom AUR packages before running "xaprepare". :joy: And instead, complete the work on recognizing those custom packages.

Reply:
Thanks for the explanation :) Yes, I think your approach to let the users take care of installation of those helpers so that we don't force anyone to use what they don't need/like/want. It's, indeed, much better to do more work on our side and adapt to the user's environment.